### PR TITLE
Add support opt-in regional STS endpoints

### PR DESCRIPTION
**Pull request by AllanFly120**
This PR makes STS client supports sending requests to regional endpoints(e.g. sts.us-west-2.amazonaws.com). You can opt in by setting `stsRegionalEndpoints` client config or `AWS_STS_REGIONAL_ENDPOINTS` environment or `sts_regional_endpoints` config file entry to 'regional'. 

##### Checklist

- [x] `npm run test` passes
- [x] `.d.ts` file is updated
- [x] changelog is added, `npm run add-change`
- [x] run `bundle exec rake docs:api` and inspect `doc/latest/index.html` if documentation is changed

### CODE REVIEW

File: lib/services/sts.js [hunk 1]
```diff
@@ -43,5 +46,87 @@ AWS.util.update(AWS.STS.prototype, {
...
+        code: 'InvalidConfiguration',
+        message: 'invalid "stsRegionalEndpoints" configuration. Expect "legacy" ' +
+        ' or "regional". Got "' + config.stsRegionalEndpoints + '".'
+      });
+    }
+    if (!AWS.util.isNode()) return;
+    //validate environmental variable
+    if (Object.prototype.hasOwnProperty.call(process.env, ENV_REGIONAL_ENDPOINT_ENABLED)) {
+      var envFlag = process.env[ENV_REGIONAL_ENDPOINT_ENABLED];
+      if (typeof envFlag === 'string' && ['legacy', 'regional'].indexOf(envFlag.toLowerCase()) >= 0) {
```
Comment (trivikr) [inline lib/services/sts.js]: This if-else be moved to a different function, as it's similar across the three blocks

**Comment by greensopinion** [comment]
In what looks like it's related to this change, we've started seeing the following when deploying with the serverless framework on our CI machines:

```
TypeError: Cannot convert undefined or null to object 
    at hasOwnProperty (<anonymous>) 
    at features.constructor.validateRegionalEndpointsFlag (/usr/local/lib/node_modules/serverless/node_modules/aws-sdk/lib/services/sts.js:92:41) 
    at features.constructor.optInRegionalEndpoint (/usr/local/lib/node_modules/serverless/node_modules/aws-sdk/lib/services/sts.js:106:10) 
    at features.constructor.validateService (/usr/local/lib/node_modules/serverless/node_modules/aws-sdk/lib/services/sts.js:124:10) 
    at features.constructor.initialize (/usr/local/lib/node_modules/serverless/node_modules/aws-sdk/lib/service.js:52:10) 
    at features.constructor.Service [as constructor] (/usr/local/lib/node_modules/serverless/node_modules/aws-sdk/lib/service.js:40:10) 
    at features.constructor (/usr/local/lib/node_modules/serverless/node_modules/aws-sdk/lib/util.js:624:24) 
    at new features.constructor (/usr/local/lib/node_modules/serverless/node_modules/aws-sdk/lib/util.js:624:24) 
    at features.constructor.Service [as constructor] (/usr/local/lib/node_modules/serverless/node_modules/aws-sdk/lib/service.js:31:17) 
    at new features.constructor (/usr/local/lib/node_modules/serverless/node_modules/aws-sdk/lib/util.js:624:24) 
    at persistentRequest (/usr/local/lib/node_modules/serverless/lib/plugins/aws/provider/awsProvider.js:268:26) 
    at doCall (/usr/local/lib/node_modules/serverless/lib/plugins/aws/provider/awsProvider.js:226:9) 
    at BbPromise (/usr/local/lib/node_modules/serverless/lib/plugins/aws/provider/awsProvider.js:243:14) 
    at persistentRequest (/usr/local/lib/node_modules/serverless/lib/plugins/aws/provider/awsProvider.js:224:38) 
    at Object.request.requestQueue.add [as promiseGenerator] (/usr/local/lib/node_modules/serverless/lib/plugins/aws/provider/awsProvider.js:264:49) 
    at Queue._dequeue (/usr/local/lib/node_modules/serverless/node_modules/promise-queue/lib/index.js:153:30) 
    at /usr/local/lib/node_modules/serverless/node_modules/promise-queue/lib/index.js:109:18 
From previous event: 
    at Queue.add (/usr/local/lib/node_modules/serverless/node_modules/promise-queue/lib/index.js:94:16) 
    at AwsProvider.request (/usr/local/lib/node_modules/serverless/lib/plugins/aws/provider/awsProvider.js:264:39) 
    at AwsProvider.getAccountInfo (/usr/local/lib/node_modules/serverless/lib/plugins/aws/provider/awsProvider.js:462:17) 
    at AwsProvider.getAccountId (/usr/local/lib/node_modules/serverless/lib/plugins/aws/provider/awsProvider.js:457:17) 
    at AwsDeploy.checkLogGroupSubscriptionFilterResourceLimitExceeded (/usr/local/lib/node_modules/serverless/lib/plugins/aws/deploy/lib/checkForChanges.js:151:26) 
    at AwsDeploy.BbPromise.bind.then.then.then.then (/usr/local/lib/node_modules/serverless/lib/plugins/aws/deploy/lib/checkForChanges.js:29:21) 
    at FSReqWrap.readFileAfterClose [as oncomplete] (internal/fs/read_file_context.js:53:3) 
From previous event: 
    at AwsDeploy.checkForChanges (/usr/local/lib/node_modules/serverless/lib/plugins/aws/deploy/lib/checkForChanges.js:23:8) 
From previous event: 
    at Object.aws:deploy:deploy:checkForChanges [as hook] (/usr/local/lib/node_modules/serverless/lib/plugins/aws/deploy/index.js:104:10) 
    at BbPromise.reduce (/usr/local/lib/node_modules/serverless/lib/classes/PluginManager.js:408:55) 
From previous event: 
    at PluginManager.invoke (/usr/local/lib/node_modules/serverless/lib/classes/PluginManager.js:408:22) 
    at PluginManager.spawn (/usr/local/lib/node_modules/serverless/lib/classes/PluginManager.js:426:17) 
    at AwsDeploy.BbPromise.bind.then (/usr/local/lib/node_modules/serverless/lib/plugins/aws/deploy/index.js:93:48) 
From previous event: 
    at Object.deploy:deploy [as hook] (/usr/local/lib/node_modules/serverless/lib/plugins/aws/deploy/index.js:89:10) 
    at BbPromise.reduce (/usr/local/lib/node_modules/serverless/lib/classes/PluginManager.js:408:55) 
From previous event: 
    at PluginManager.invoke (/usr/local/lib/node_modules/serverless/lib/classes/PluginManager.js:408:22) 
    at PluginManager.run (/usr/local/lib/node_modules/serverless/lib/classes/PluginManager.js:439:17) 
    at variables.populateService.then (/usr/local/lib/node_modules/serverless/lib/Serverless.js:109:33) 
    at runCallback (timers.js:705:18) 
    at tryOnImmediate (timers.js:676:5) 
    at processImmediate (timers.js:658:5) 
    at process.topLevelDomainCallback (domain.js:121:23) 
From previous event: 
    at Serverless.run (/usr/local/lib/node_modules/serverless/lib/Serverless.js:96:6) 
    at serverless.init.then (/usr/local/lib/node_modules/serverless/bin/serverless:43:28) 
    at /usr/local/lib/node_modules/serverless/node_modules/graceful-fs/graceful-fs.js:136:16 
    at /usr/local/lib/node_modules/serverless/node_modules/graceful-fs/graceful-fs.js:57:14 
    at FSReqWrap.oncomplete (fs.js:141:20) 
```

It's not clear to me if this is a new defect, or something wrong with our use of the AWS SDK.
It works fine with version 2.457.0 but fails with 2.525.0

**Comment by greensopinion** [comment]
Updating the serverless CLI from 1.41.1 to 1.51.0 has resolved the issue.

**Comment by greensopinion** [comment]
> Updating the serverless CLI from 1.41.1 to 1.51.0 has resolved the issue.

Actually, it has only resolved the issue for those that have a `.aws/config` with a `[default]` section

**Pull request merged**